examples/pytorch/schnet/Alchemy_dataset.py

```diff
@@ -0,0 +1,280 @@
+# -*- coding:utf-8 -*-
+"""Example dataloader of Tencent Alchemy Dataset
+https://alchemy.tencent.com/
+"""
+import os
+import zipfile
+import os.path as osp
+from rdkit import Chem
+from rdkit.Chem import ChemicalFeatures
+from rdkit import RDConfig
+import dgl
+from dgl.data.utils import download
+import torch
+from collections import defaultdict
+from torch.utils.data import Dataset
+from torch.utils.data import DataLoader
+import pathlib
+import pandas as pd
+import numpy as np
+_urls = {'Alchemy': 'https://alchemy.tencent.com/data/'}
+
+
+class AlchemyBatcher:
+    def __init__(self, graph=None, label=None):
+        self.graph = graph
+        self.label = label
+
+
+def batcher():
+    def batcher_dev(batch):
+        graphs, labels = zip(*batch)
+        batch_graphs = dgl.batch(graphs)
+        labels = torch.stack(labels, 0)
+        return AlchemyBatcher(graph=batch_graphs, label=labels)
+
+    return batcher_dev
+
+
+class TencentAlchemyDataset(Dataset):
+
+    fdef_name = osp.join(RDConfig.RDDataDir, 'BaseFeatures.fdef')
+    chem_feature_factory = ChemicalFeatures.BuildFeatureFactory(fdef_name)
+
+    def alchemy_nodes(self, mol):
+        """Featurization for all atoms in a molecule. The atom indices
+        will be preserved.
+
+        Args:
+            mol : rdkit.Chem.rdchem.Mol
+              RDKit molecule object
+        Returns
+            atom_feats_dict : dict
+              Dictionary for atom features
+        """
+        atom_feats_dict = defaultdict(list)
+        is_donor = defaultdict(int)
+        is_acceptor = defaultdict(int)
+
+        fdef_name = osp.join(RDConfig.RDDataDir, 'BaseFeatures.fdef')
+        mol_featurizer = ChemicalFeatures.BuildFeatureFactory(fdef_name)
+        mol_feats = mol_featurizer.GetFeaturesForMol(mol)
+        mol_conformers = mol.GetConformers()
+        assert len(mol_conformers) == 1
+        geom = mol_conformers[0].GetPositions()
+
+        for i in range(len(mol_feats)):
+            if mol_feats[i].GetFamily() == 'Donor':
+                node_list = mol_feats[i].GetAtomIds()
+                for u in node_list:
+                    is_donor[u] = 1
+            elif mol_feats[i].GetFamily() == 'Acceptor':
+                node_list = mol_feats[i].GetAtomIds()
+                for u in node_list:
+                    is_acceptor[u] = 1
+
+        num_atoms = mol.GetNumAtoms()
+        for u in range(num_atoms):
+            atom = mol.GetAtomWithIdx(u)
+            symbol = atom.GetSymbol()
+            type = atom.GetAtomicNum()
```

Comment:
Sorry this is my bad. `type` should be reserved and you can replace it by another name, E.g. `atom_type`.

Reply:
Gotcha.